System/ContentFile.h

```diff
@@ -32,483 +19,253 @@ struct ga_Handle;
+		/// <returns>An error return value signaling success or any particular failure. Anything below 0 is an error signal.</returns>
+		int Create(const ContentFile &reference);
+#pragma endregion
+
+#pragma region Destruction
+		/// <summary>
+		/// Destructor method used to clean up a ContentFile object before deletion from system memory.
+		/// </summary>
+		virtual ~ContentFile() { Destroy(true); }
+
+		/// <summary>
+		/// Destroys and resets (through Clear()) the ContentFile object.
+		/// </summary>
+		/// <param name="notInherited">Whether to only destroy the members defined in this derived class, or to destroy all inherited members also.</param>
+		virtual void Destroy(bool notInherited = false) { Clear(); }
+
+		/// <summary>
+		/// Resets the entire ContentFile, including its inherited members, to their default settings or values.
+		/// </summary>
+		virtual void Reset() { Clear(); }
+
+		/// <summary>
+		/// Frees all loaded data used by all ContentFile instances. This should ONLY be done when quitting the app, or after everything else is completely destroyed.
+		/// </summary>
+		static void FreeAllLoaded();
+#pragma endregion
+
+#pragma region INI Handling
+		/// <summary>
+		/// Reads a property value from a Reader stream. If the name isn't recognized by this class, then ReadProperty of the parent class is called.
+		/// If the property isn't recognized by any of the base classes, false is returned, and the Reader's position is untouched.
+		/// </summary>
+		/// <param name="propName">The name of the property to be read.</param>
+		/// <param name="reader">A Reader lined up to the value of the property to be read.</param>
+		/// <returns>
+		/// An error return value signaling whether the property was successfully read or not.
+		/// 0 means it was read successfully, and any nonzero indicates that a property of that name could not be found in this or base classes.
+		/// </returns>
+		virtual int ReadProperty(std::string propName, Reader &reader);
+
+		/// <summary>
+		/// Saves the complete state of this ContentFile to an output stream for later recreation with Create(Reader &reader).
+		/// </summary>
+		/// <param name="writer">A Writer that the ContentFile will save itself with.</param>
+		/// <returns>An error return value signaling success or any particular failure. Anything below 0 is an error signal.</returns>
+		virtual int Save(Writer &writer) const;
+#pragma endregion
+
+#pragma region Getters and Setters
+		/// <summary>
+		/// Gets the class name of this Entity.
+		/// </summary>
+		/// <returns>A string with the friendly-formatted type name of this object.</returns>
+		virtual const std::string & GetClassName() const { return m_ClassName; }
+
+		/// <summary>
+		/// Gets the ID of the Data Module this file is inside.
+		/// </summary>
+		/// <returns>The ID of the Data Module containing this' file.</returns>
+		int GetDataModuleID();
+
+		/// <summary>
```

Comment:
I'm not sure we need the commented datasize method. I'm quite sure we don't need the allegro datafile one, since contentfile isn't just allegro data (it can be sounds and junk). Or am I misunderstanding and this is for if it was a packed allegro .dat file or whatever?

Reply:
This is for datafile routines.

Reply:
And we're keeping those around for in case we decide to start packing base rtes as datafiles right?

Reply:
Yes.